src/piadm.sh

```diff
@@ -494,15 +489,16 @@ update_boot_sectors() {
 			fi
 			# otherwise mount the ESP and trash it.
 			tdir=$(mktemp -d)
-			mount -F pcfs /dev/dsk/${a}s0 ${tdir}
-			if [[ $? -ne 0 ]]; then
+			if mount -F pcfs "/dev/dsk/${a}s0" "${tdir:?}" ; then
+				some=1
+			else
 				eecho "disk $a has no PCFS ESP, it seems"
 				continue
 			fi
 			# Just take out the EFI directory, in case someone
 			# is using it for something ELSE also.
-			/bin/rm -rf ${tdir}/EFI
-			umount ${tdir} && rmdir "$tdir"
+			/bin/rm -rf "${tdir}/EFI"
+			umount "$tdir" && rmdir "$tdir"
 			# If we make it here, at least some disks had
 			# ESP and we managed to clean them out.  "some" below
 			# will get set.
```

Comment:
This comment should be modified, and right after "some" should be set.

Reply:
I juggled this around a little bit. I think it should work for what is intended.